irc/util.go

```diff
@@ -2,43 +2,64 @@ package irc
 
 import "fmt"
 
+// User represents a user identified by a nick and an ident
+type User struct {
+	Nick  string
+	Ident string
+	Host  string
+}
+
+func (u User) String() string {
+	if len(u.Nick) > 0 && len(u.Ident) > 0 {
+		if len(u.Host) > 0 {
+			return fmt.Sprintf("%v!%v@%v", u.Nick, u.Ident, u.Host)
+		}
+		return fmt.Sprintf("%v!%v@localhost", u.Nick, u.Ident)
+	}
+	return ""
+}
+
 // Privmsg represents a line in an IRC conversation
 type Privmsg struct {
-	From    string
+	From    User
 	Channel string
 	Message string
 }
 
 // ToMessage turns a Privmsg into a Message
 func (p *Privmsg) ToMessage() *Message {
-	var prefixStr string
-	if p.From != "" {
-		prefixStr = fmt.Sprintf("%v!%[1]v@localhost", p.From)
-	}
-
 	return &Message{
-		prefixStr,
+		p.From.String(),
 		PrivmsgCmd,
 		[]string{p.Channel, p.Message},
 	}
 }
 
 // Nick represents a IRC user nick change event
 type Nick struct {
-	From    string
+	From    User
 	NewNick string
 }
 
 // ToMessage turns a Nick into a Message
 func (n *Nick) ToMessage() *Message {
-	var prefixStr string
-	if n.From != "" {
-		prefixStr = fmt.Sprintf("%v!%[1]v@localhost", n.From)
-	}
-
 	return &Message{
-		prefixStr,
+		n.From.String(),
 		NickCmd,
 		[]string{n.NewNick},
 	}
 }
+
+// Pong is a PONG message
+type Pong struct {
+	Token string
+}
+
+// ToMessage turns a Pong into a Message
+func (p *Pong) ToMessage() *Message {
+	return &Message{
+		"tanya",
```

Comment:
isn't this a variable somewhere

Reply:
it should probably be configurable (it would be something like `irc.indojin.org` in a real IRC server), but no it's not a variable anywhere